chore(docs): fixing broken links in the docs (#5340)
winglang/wing: docs/contributing/998-archived/01-compile-targets.md

> Compiler plugins serve to extend the capabilities of the Wing compiler. They offer customization for the compilation output, such as infrastructure definitions. For instance, a compiler plugin can be employed to enforce a specific tagging convention on all resources within the resulting infrastructure.

Reviewer: ```suggestion
Platforms extend the capabilities of the Wing compiler. They offer customization for the compilation output, such as infrastructure definitions. For instance, a platform can enforce a specific tagging convention on all resources within the resulting infrastructure.
```

Author: This is part of an archived doc about compilation targets, so I don't think we should update the text in it. The reason I'm still updating the link is because it is broken and we have an automatic tool that notifies us about broken links, so if I won't fix this link it will add noise.

> Learn more about compiler plugins [here](https://www.winglang.io/contributing/archived/compiler-plugins)

Reviewer: ```suggestion
Learn more about platforms [here](https://www.winglang.io/docs/concepts/platforms)
```

Author: Same remark as above. This is an archived doc.
